distributed/chaos.py

```diff
@@ -0,0 +1,44 @@
+import asyncio
+import random
+import sys
+
+from dask.utils import parse_timedelta
+
+from distributed.diagnostics.plugin import WorkerPlugin
+
+
+class KillWorker(WorkerPlugin):
+    def __init__(self, delay="100 s", mode="sys.exit"):
+        self.delay = parse_timedelta(delay)
+        if mode not in ("sys.exit", "graceful", "segfault"):
+            raise ValueError(
+                f"Three modes supported, 'sys.exit', 'graceful', and 'segfault'. "
+                f"got {mode}"
+            )
+        self.mode = mode
+
+    async def setup(self, worker):
+        self.worker = worker
+        if self.mode == "graceful":
+            f = self.graceful
+        elif self.mode == "sys.exit":
+            f = self.sys_exit
+        elif self.mode == "segfault":
+            f = self.segfault
+
+        self.worker.loop.asyncio_loop.call_later(
+            delay=random.expovariate(1 / self.delay),
```

Comment:
> Returned values range from 0 to positive infinity if lambd is positive

Might be nice to bound that to something sane (say, no more than twice as high or low as the target value).

```python
In [14]: [random.expovariate(1/10) for _ in range(10)]
Out[14]: 
[13.091022644736674,
 3.4401150413923056,
 1.176538629303591,
 4.343163380659492,
 21.167619469980696,
 4.6599991211540885,
 5.4169792796804295,
 3.2745027172741246,
 35.03846504902826,
 13.781649481654647]
```
 I'm not sure a range of 1.17. to 35 sec would be ideal if you set 10s as the target. All for chaos, but you may want your chaos to be a little bit more predictable.

Reply:
I'm inclined to keep this as-is for now.  I feel like this is the most boring choice, and also represents physical stuff that fails with constant probability at every time step.  There won't be any pattern to failures.

Reply:
Resolving this.  I'm going to rely on the poisson process argument here as being the simplest possible choice.  In the future folks can deviate from this if desired.